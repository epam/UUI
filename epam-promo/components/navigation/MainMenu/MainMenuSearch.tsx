import * as React from 'react';
import { TextInputProps, TextInput } from '@epam/uui-components';
import { ReactComponent as LensIcon } from '../../../icons/search-18.svg';
import { ReactComponent as CancelIcon } from '../../../icons/menu_input_cancel.svg';
import { cx, IAdaptiveItem, IEditableDebouncer } from '@epam/uui';
import * as css from './MainMenuSearch.scss';

export interface MainMenuSearchProps extends TextInputProps, IAdaptiveItem {
}

export const MainMenuSearch = (props: MainMenuSearchProps) => (
    <IEditableDebouncer
        { ...props }
        render={ iEditable => <TextInput
            iconPosition='left'
            icon={ LensIcon }
            cancelIcon={ props.value?.length > 0 && CancelIcon }
            { ...props }
            { ...iEditable }
            cx={ cx(css.searchInput, props.cx) }
        /> }
    />

);
