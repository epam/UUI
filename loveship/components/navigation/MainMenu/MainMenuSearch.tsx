import React from 'react';
import { TextInputProps, TextInput } from '@epam/uui-components';
import { ReactComponent as LensIcon } from '../../icons/search-18.svg';
import { ReactComponent as CancelIcon } from '../../icons/menu_input_cancel.svg';
import { cx, IAdaptiveItem } from '@epam/uui-core';
import css from './MainMenuSearch.scss';

export interface MainMenuSearchProps extends TextInputProps, IAdaptiveItem {}

export const MainMenuSearch = React.forwardRef<HTMLInputElement, MainMenuSearchProps>((props: MainMenuSearchProps, ref) => (
    <TextInput
        ref={ref}
        iconPosition="left"
        icon={LensIcon}
        cancelIcon={props.value.length > 0 && CancelIcon}
        {...props}
        cx={cx(css.searchInput, props.cx)}
    />
));

MainMenuSearch.displayName = 'MainMenuSearch';
