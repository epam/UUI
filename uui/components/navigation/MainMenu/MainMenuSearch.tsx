import * as React from 'react';
import { TextInputProps, TextInput } from '@epam/uui-components';
import { ReactComponent as LensIcon } from '@epam/assets/icons/action-search-outline.svg';
import { ReactComponent as CancelIcon } from '@epam/assets/icons/navigation-close-outline.svg';
import { cx, IAdaptiveItem, IEditableDebouncer } from '@epam/uui-core';
import css from './MainMenuSearch.module.scss';

/** Represents the properties of the MainMenuSearch component. */
export interface MainMenuSearchProps extends TextInputProps, IAdaptiveItem {}

export const MainMenuSearch = /* @__PURE__ */React.forwardRef<HTMLInputElement, MainMenuSearchProps>((props, ref) => (
    <IEditableDebouncer
        { ...props }
        render={ (iEditable) => (
            <TextInput
                iconPosition="left"
                icon={ LensIcon }
                cancelIcon={ props.value?.length > 0 && CancelIcon }
                { ...props }
                { ...iEditable }
                ref={ ref }
                cx={ cx(css.searchInput, props.cx) }
                { ...props.rawProps }
            />
        ) }
    />
));
