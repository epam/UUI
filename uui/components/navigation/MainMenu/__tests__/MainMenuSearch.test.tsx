import React from 'react';
import { MainMenuSearch } from '../MainMenuSearch';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';

describe('MainMenuSearch', () => {
    it('should be rendered correctly with null value', async () => {
        const tree = await renderSnapshotWithContextAsync(<MainMenuSearch value={ undefined } onValueChange={ jest.fn } />);

        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with value and extra props', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <MainMenuSearch
                value="test"
                onValueChange={ jest.fn }
                type="primary"
                collapseToMore
                estimatedWidth={ 120 }
                showInBurgerMenu
                priority={ 6 }
            />,
        );

        expect(tree).toMatchSnapshot();
    });
});
