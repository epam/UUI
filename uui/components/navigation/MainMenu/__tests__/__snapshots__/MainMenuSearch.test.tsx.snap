// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuSearch should be rendered correctly with null value 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box -clickable searchInput"
    tabindex="-1"
  >
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
    <input
      class="uui-input"
      tabindex="0"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`MainMenuSearch should be rendered correctly with value and extra props 1`] = `
<DocumentFragment>
  <div
    class="container uui-input-box -clickable searchInput"
    tabindex="-1"
  >
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
    <input
      class="uui-input"
      tabindex="0"
      type="primary"
      value="test"
    />
  </div>
</DocumentFragment>
`;
